libqfieldsync/project_checker.py

```diff
@@ -439,3 +445,43 @@ def check_external_layers(
             )
 
         return None
+
+    def check_project_layers_sources_actions(self) -> Optional[FeedbackResult]:
+        """Check if layers from the same GeoPackage have offline and copy actions."""
+        layer_sources_by_filename: dict[str, list[LayerSource]] = defaultdict(list)
+
+        for project_layer in self.project.mapLayers().values():
+            layer_source = LayerSource(project_layer)
+            if not layer_source.is_file:
+                continue
+
+            if layer_source.action not in [SyncAction.OFFLINE, SyncAction.COPY]:
+                continue
+
+            layer_sources_by_filename[layer_source.filename].append(
+                (layer_source.name, layer_source.action)
+            )
```

Comment:
```suggestion
            layer_sources_by_filename[layer_source.filename].append(layer_source)
```
The code before does not match the type definition on line 451. I believe this is a bit more readable, avoid keyless datastructures for data where ordering matters.